### ISSUE TITLE
Install script set my shell to /usr/local/bin/zsh instead of /bin/zsh

### ISSUE THREAD

**Issue by carc1n0gen** [robbyrussell/oh-my-zsh#5277]
The installer set my shell to `/usr/local/bin/zsh`, but I don't even have brew installed and that path doesn't exist.

I was able to fix this by manually settings my shell to /bin/bash in terminal preferences, then running `chsh -s /bin/zsh` and everything is working as expected now.

I should also mention that I am using macOS Sierra beta2

**Comment by apjanke** [comment]
Have a look at your `/etc/shells` file. Is `/usr/local/bin/zsh` listed there? That file is used by the OMZ installer to determine what shells are installed.

**Comment by carc1n0gen** [comment]
Well would you look at that:

```
# List of acceptable shells for chpass(1).
# Ftpd will not allow users to connect who are not using
# one of these shells.

/bin/bash
/bin/csh
/bin/ksh
/bin/sh
/bin/tcsh
/bin/zsh
/usr/local/bin/fish
/usr/local/bin/zsh
```

That is weird, since this was a fresh install and I never edited that file.  Looked at my other partition which El Capitan, looks like the installer copied the the one from there, which I had edited.

**Comment by carc1n0gen** [comment]
Looks like the `/bin/zsh` exists before `/usr/local/bin/zsh`.  Is it intended that the installer uses the last entry ending with `zsh`?

**Comment by apjanke** [comment]
> Is it intended that the installer uses the last entry ending with zsh?

Yes, but I think that is simply because it was easier to code. I don't know if there's a convention for prioritizing shells by order in `/etc/shells`; I don't think its order is significant.

**Comment by apjanke** [comment]
The fix for this is probably to make the `zsh` detection code more robust: go through the list of `zsh`s in `/etc/shells` and choose the first (or last) one that exists, and bail if none of them exist. That feels a little hinky to me, though, because it's working around a possibly-bad local configuration.

**Comment by v0idst4r** [comment]
#6398 should fix this.

**Comment by mcornella** [comment]
The installer in #5169 now has a better way to get the zsh path to change to: first it uses the first one that appears in `$PATH` and checks it's in /etc/shells; otherwise it gets one from /etc/shells and it checks that it really exists, so you won't have the problem of having a default shell that doesn't exist. Hopefully 🤞 

If you want to help speed up the testing and merge (which should happen early next week), you can wget or curl it from the URL `https://raw.githubusercontent.com/mcornella/oh-my-zsh/refactor-installer/tools/install.sh`. Read the comments at the top of the script to know how to call it.

Thanks!

**Comment by mcornella** [comment]
Closed by #5169.
